Docs: Update datasource API examples (#20951)
grafana/grafana: docs/sources/http_api/data_source.md

{"id":1,"message":"Datasource added", "name": "test_datasource"}
```

> > NOTE: `password` and `basicAuthPassword` should be defined under `secureJsonData` in order to be stored in an encrypted blob in the database and no more sent to the browser. Check also the [Encrypting Sensitive Data]({{< relref "../plugins/developing/auth-for-datasources.md/#encrypting-sensitive-data">}}) documentation for more details.

Reviewer: Does it make sense to include `password` and `basicAuthPassword` in `secureJsonData` as part of your example?

> > NOTE: `password` and `basicAuthPassword` should be defined under `secureJsonData` in order to be stored in an encrypted blob in the database and no more sent to the browser. Check also the [Encrypting Sensitive Data]({{< relref "../plugins/developing/auth-for-datasources.md/#encrypting-sensitive-data">}}) documentation for more details.

Reviewer 2: ```suggestion
> NOTE: `password` and `basicAuthPassword` should be defined under `secureJsonData` in order to be stored securely as an encrypted blob in the database. See the [Encrypting Sensitive Data]({{< relref "../plugins/developing/auth-for-datasources.md/#encrypting-sensitive-data">}}) documentation for more details.
```

{"message":"Datasource updated", "id": 1, "name": "test_datasource"}
```

> > NOTE: `password` and `basicAuthPassword` should be defined under `secureJsonData` in order to be stored in an encrypted blob in the database and no more sent to the browser. Check also the [Encrypting Sensitive Data]({{< relref "../plugins/developing/auth-for-datasources.md/#encrypting-sensitive-data">}}) documentation for more details.

Reviewer: I guess the example above should be updated, since it doesn't use `secureJsonData` and also includes `password` and `basicAuthPassword` directly?

}
```

> 

Reviewer: Why is the example response removed?

Author: it's not removed, I have voted it before the basic auth request and response example [here](https://github.com/grafana/grafana/blob/016c4a1d06c1e409f19a97ca29c3468e7a2e2c03/docs/sources/http_api/data_source.md#L195).

Reviewer: I don't really understand what you mean, to be honest :/ When I check the diff vs master, the example Cloudwatch response is removed in your branch.

  "database":"",
  "basicAuth":true,
  "basicAuthUser":"basicuser",

>   "basicAuthPassword":"",

Reviewer: Should `.basicAuthpassword` be part of the request?

Author: correct it should be omitted.
